navigator.go

```diff
@@ -99,8 +100,7 @@ func (n navigator) Follow(rel string) navigator {
 // Followf adds a relation to the follow queue of the navigator, with a
 // set of parameters to expand on execution.
 func (n navigator) Followf(rel string, params P) navigator {
-	relations := make([]relation, 0, len(n.path)+1)
-	copy(n.path, relations)
```

Comment:
This is the problem.  This doesn't do what you expect here.  So I copied the array a different way (starting from an empty one and then appending all existing elements).  This is probably not as efficient, but it works. :-)